Bug fix: wrong current working directory

kinzlaw opened the pull request:
We faced `reorganize_markers.py killed` errors several times, without any clue of what's wrong. Then I tried `bash .command.run` with customized `.command.sh` to get into the interactive docker, and found out that reorganize_markers.py was running under the root directory.(I've added `docker.runOptions = '-u $(id -u):$(id -g)'` to nextflow.config, don't know if it's related to this bug)

Guess there would be better solution to change the working directory with nextflow's code, but I'm not familiar with this language, and don't know if my code is suitable. Please make a decision at your will, thanks.

sminot commented:
The nextflow command launcher should be managing the process in order to set the current working directory appropriately. I would be rather surprised if this change would fix the error that you're seeing. Can you clarify whether you've tested this updated code on your end in the context of the larger workflow, or just in the manual run of a single process with your modified docker entry script?

kinzlaw commented:
Yes, I've tested the code with the 350-genomes.

Sam Minot ***@***.***> 于 2022年4月5日周二 01:53写道：

> The nextflow command launcher should be managing the process in order to
> set the current working directory appropriately. I would be rather
> surprised if this change would fix the error that you're seeing. Can you
> clarify whether you've tested this updated code on your end in the context
> of the larger workflow, or just in the manual run of a single process with
> your modified docker entry script?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FredHutch/gig-map/pull/7#issuecomment-1087845732>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABC5ECFLFCHOCK5KQ5ESKQLVDMUCJANCNFSM5SQD4FRQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

sminot commented:
Would you mind sharing the log files from (a) the previous run which failed and (b) the updated run with the code from the PR? Please feel free to share by email if that is easier.

kinzlaw commented:
The log files for this problem just got very messy(and I just found that the relating `.nextflow.log` has been overwritten by later runs), since we've been tackling with shortage of hardware(32-64cores and 256GB memory), we've changed our nextflow.config several times during the process(to deal with lack of memory errors, and to help maximize the CPU/memory utilization). So there're a lot of *.report.html and .nextflow.log files.

Let me try to do some digest, all the following are related to the 350-NCBI-genomes and 7 local fasta run.
- At first, `align.sh` just got crashed with every time, with an error message similar to the following(the screenshot is related to out of memory error, which I've fixed with modifying nextflow.config to enlarge the memory param). No detail of the error is logged(`print()` in python script can't get to the log file).
![image](https://user-images.githubusercontent.com/4575752/161656492-dd96a1c6-0a6e-474e-92f6-e4f1a97b8d3a.png)

- I added `sleep 100000` to .command.sh, and successfully ran the python script in an interactive docker environment, and found that the working directory is wrong. So added the code in this PR, then reorganize_markers.py worked as expected.

I think that adding this code won't harm even if the bug doesn't exist, and please consider add `pwd` to my modified location of the bash script(it will probably be logged in the .nextflow.log), and see if it really is my case in your running environment.

kinzlaw commented:
BTW, as a test, I've run through the whole workflow with a 14-genomes NCBI download with unmodified code, successful without reorganize_markers.py being killed. Maybe this python script isn't always called?

sminot commented:
I wasn't able to reproduce this bug sufficiently to merge before implementing the refactor described in #6 . 

Please refer to that issue and try out the new implementation, and feel free to open another PR if you are able to reproduce this bug in the new version and find a good fix for it.
